scripts/core/CONTRIB.rst

```diff
@@ -1,71 +1,204 @@
+are generated from a set of YAML_ files which are used as inputs to a Mako_
+based templating system. The YAML file syntax is defined in `YAML syntax`_. To
+generate the outputs of the Mako templates a build directory must be
+configured, instructions are available in the `README
+<https://github.com/oneapi-src/unified-runtime/blob/main/README.md>`_ file.
+Upon successfully configuring a build directory, generate the outputs with the
+following command (or suitable build system equivalent):
 
-    * Entry-Points use ``camelCase`` conventions i.e ``${x}DeviceGet``.
+.. code-block:: console
 
-    * Argument names are also `camelCase` and are prefixed with a type 
-      indicator:
+    $ cmake --build build --target generate
 
-        * Pointer arguments are prefixed with ``p`` or ``pp`` for double pointer 
-          i.e. ``pEvent``.
+.. _YAML: https://yaml.org/
+.. _Mako: https://www.makotemplates.org/
+.. _YAML syntax:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/YaML.md
 
-        * Handle arguments are prefixed with ``h`` i.e. ``hQueue``.
+Writing YAML
+============
 
+Please read the :ref:`core/INTRO:Naming Convention` section prior to making a
+contribuition and refer to the `YAML syntax`_ for specifics of how to define
+the required contructs.
 
+When writing ``*.yml`` files and ``ur`` or ``UR`` should exist in the output
+use ``$${'x'}`` or ``$${'X'}`` respectively. These will be replaced while
+`Generating Source`_.
 
+Additionally, the following conventions *must* be followed for function
+arguments:
+
+*   Argument names are ``camelCase``.
+*   Arguments with pointer types are prefixed with ``p`` for each pointer in
+    the type i.e. ``char *pMessage``, ``char **ppMessage``, etc.
+*   Handle arguments are prefixed with ``h`` i.e. ``hQueue``.
+*   Pointer to handle arguments, such as out parameters, are prefix with ``ph``
+    i.e. ``phQueue``.
 
 Forks and Pull Requests
 =======================
 
-To submit a pull request to Unified Runtime, you must first create your own 
-personal fork of the project and submit your changes to a branch. By convention 
-we name our branches ``<your_name>/<short_description>``, where the description 
-indicates the intent of your change. You can then raise a pull request targeting 
-``oneapi-src/unified-runtime:main``.
+To submit a pull request to Unified Runtime, you must first create your own
+personal fork of the project and submit your changes to a branch. By convention
+we name our branches ``<your_name>/<short_description>``, where the description
+indicates the intent of your change. You can then raise a pull request
+targeting ``oneapi-src/unified-runtime:main``.
+
+When making changes to the specification you *must* commit all changes to files
+in the repository as a result of `Generating Source`_.
+
+Before your pull request is merged it *must* pass all jobs in the GitHub
+Actions workflow and *must* be reviewed by no less than two code owners.
+
+.. hint::
```

Comment:
For me this is not rendering correctly?

Reply:
I may be worth force force refreshing for browser.